stdlib/LinearAlgebra/src/symmetriceigen.jl

```diff
@@ -164,26 +164,44 @@ function eigen!(A::Hermitian{T,S}, B::Hermitian{T,S}; sortby::Union{Function,Not
     vals, vecs, _ = LAPACK.sygvd!(1, 'V', A.uplo, A.data, B.uplo == A.uplo ? B.data : copy(B.data'))
     GeneralizedEigen(sorteig!(vals, vecs, sortby)...)
 end
-function eigen!(A::RealHermSymComplexHerm{T,<:StridedMatrix}, B::AbstractMatrix{T}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
-    return _choleigen!(A, B, sortby)
+function eigen(A::RealHermSymComplexHerm{T,<:StridedMatrix}, B::AbstractMatrix{T}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    return eigen!(Matrix{T}(A), eigencopy_oftype(B, T); sortby)
 end
-function eigen!(A::StridedMatrix{T}, B::Union{RealHermSymComplexHerm{T},Diagonal{T}}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
-    return _choleigen!(A, B, sortby)
+function eigen(A::StridedMatrix{T}, B::Union{RealHermSymComplexHerm{T},Diagonal{T}}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
+    return eigen!(eigencopy_oftype(A, T), Matrix{T}(B); sortby)
 end
-function _choleigen!(A, B, sortby)
-    U = cholesky(B).U
-    vals, w = eigen!(UtiAUi!(A, U))
-    vecs = U \ w
+
+function eigen(A::AbstractMatrix{T}, C::Cholesky{T, <:AbstractMatrix}; sortby::Union{Function,Nothing}=nothing) where {T<:Number}
```

Comment:
In the Cholesky-case, I do think we had an in-place version, hadn't we? If so, then it would make sense to have the `eigen`/`eigen!` split. It's the BunchKaufman-case where we couldn't perform the algorithm in-place.

Reply:
@dkarrasch Sorry! I misunderstood your intentions. I think I've now reverted the removal of the in-place functions in commit https://github.com/aravindh-krishnamoorthy/julia/commit/3696cfcb350efc70427e7b907d05ea24cb255261.

Please do let me know in case you have further comments. I do not mind fixing minor things at all! If in the end, the code is good and useful, I'm happy. Danke!